src/cdk-experimental/radio/radio.ts

```diff
@@ -0,0 +1,199 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {
+  afterRenderEffect,
+  booleanAttribute,
+  computed,
+  contentChildren,
+  Directive,
+  ElementRef,
+  inject,
+  input,
+  linkedSignal,
+  model,
+  signal,
+  WritableSignal,
+} from '@angular/core';
+import {RadioButtonPattern, RadioGroupPattern} from '../ui-patterns';
+import {Directionality} from '@angular/cdk/bidi';
+import {_IdGenerator} from '@angular/cdk/a11y';
+
+// TODO: Move mapSignal to it's own file so it can be reused across components.
+
+/**
+ * Creates a new writable signal (signal V) whose value is connected to the given original
+ * writable signal (signal T) such that updating signal V updates signal T and vice-versa.
+ *
+ * This function establishes a two-way synchronization between the source signal and the new mapped
+ * signal. When the source signal changes, the mapped signal updates by applying the `transform`
+ * function. When the mapped signal is explicitly set or updated, the change is propagated back to
+ * the source signal by applying the `reverse` function.
+ */
+export function mapSignal<T, V>(
+  originalSignal: WritableSignal<T>,
+  operations: {
+    transform: (value: T) => V;
+    reverse: (value: V) => T;
+  },
+) {
+  const mappedSignal = linkedSignal(() => operations.transform(originalSignal()));
```

Comment:
wouldn't alex's suggestion using computed here make more sense? that would avoid needing to manually set both signals

Reply:
I thought that recommendation was for something that the FW would roll out. Specifically, the use of the `SIGNAL` api in the components repo. Trying to reference `SIGNAL` from the components repo auto-completes the import to `@angular/core/graph.d-BcIOep_B` which seemed like a red flag to me